0.4.1 | OSX .DS_Store

**Description:**
I tried running 0.4.1 but application startup failed

**Steps to reproduce:**

1. Download version 0.4.1
2. Create bot-config.json
3. Change config.properties
4. Running .jar via terminal

**Stacktrace :**
`org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'main': Invocation of init method failed; nested exception is java.lang.IllegalArgumentException: No save file found for name: .DS_Store
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:136)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:408)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1570)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:545)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:772)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:839)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:538)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:760)
	at org.springframework.boot.SpringApplication.createAndRefreshContext(SpringApplication.java:360)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:306)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1185)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1174)
	at ink.abb.pogo.scraper.MainKt.main(Main.kt:106)
Caused by: java.lang.IllegalArgumentException: No save file found for name: .DS_Store
	at ink.abb.pogo.scraper.services.BotRunService.load(BotRunService.kt:92)
	at ink.abb.pogo.scraper.Main.createBot(Main.kt:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:354)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:305)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:133)
	... 17 common frames omitted`

**Version:**
0.4.1

--- a/src/main/kotlin/ink/abb/pogo/scraper/Main.kt
+++ b/src/main/kotlin/ink/abb/pogo/scraper/Main.kt
@@ -12,6 +12,8 @@ import POGOProtos.Enums.PokemonIdOuterClass.PokemonId
 import POGOProtos.Inventory.Item.ItemIdOuterClass.ItemId
 import ink.abb.pogo.scraper.services.BotRunService
 import ink.abb.pogo.scraper.util.Log
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.SpringApplication
 import org.springframework.boot.autoconfigure.SpringBootApplication
@@ -19,6 +21,7 @@ import java.io.File
 import java.util.Base64
 import java.util.Properties
 import javax.annotation.PostConstruct
+import kotlin.reflect.companionObject
 
 /**
  * Allow all certificate to debug with https://github.com/bettse/mitmdump_decoder
@@ -124,6 +127,18 @@ open class Main {
             Log.red("$settingString is invalid, defaulting to $default: ${e.message}")
             return default
         }
+    }
+}
+
+fun <R : Any> R.logger(): Lazy<Logger> {
+    return lazy { LoggerFactory.getLogger(unwrapCompanionClass(this.javaClass).name) }
+}
+
+fun <T: Any> unwrapCompanionClass(ofClass: Class<T>): Class<*> {
+    return if (ofClass.enclosingClass != null && ofClass.enclosingClass.kotlin.companionObject?.java == ofClass) {
+        ofClass.enclosingClass
+    } else {
+        ofClass
     }
 }
 

--- a/src/main/kotlin/ink/abb/pogo/scraper/services/BotRunService.kt
+++ b/src/main/kotlin/ink/abb/pogo/scraper/services/BotRunService.kt
@@ -7,6 +7,7 @@ import com.pokegoapi.auth.GoogleLogin
 import com.pokegoapi.auth.PtcLogin
 import ink.abb.pogo.scraper.Bot
 import ink.abb.pogo.scraper.Settings
+import ink.abb.pogo.scraper.logger
 import ink.abb.pogo.scraper.util.Log
 import okhttp3.OkHttpClient
 import org.springframework.beans.factory.annotation.Autowired
@@ -19,6 +20,8 @@ import kotlin.concurrent.thread
  */
 @Service
 class BotRunService {
+
+    val log by logger()
 
     @Autowired
     lateinit var http: OkHttpClient
@@ -89,13 +92,13 @@ class BotRunService {
     fun load(name: String): Settings {
         val save = File(root, "$name.json")
         if(!save.isFile) {
-            throw IllegalArgumentException("No save file found for name: $name")
+            log.error("No save file found for name: $name")
         }
 
         return mapper.readValue(save, Settings::class.java)
     }
 
     fun getSaveNames(): List<String> {
-        return root.list().map { it.replace(Regex("\\.json$"), "") }
+        return root.list().filter { it.endsWith(".json") }.map { it.replace(Regex("\\.json$"), "") }
     }
 }

--- a/src/main/kotlin/ink/abb/pogo/scraper/services/BotRunService.kt
+++ b/src/main/kotlin/ink/abb/pogo/scraper/services/BotRunService.kt
@@ -92,7 +92,7 @@ class BotRunService {
     fun load(name: String): Settings {
         val save = File(root, "$name.json")
         if(!save.isFile) {
-            log.error("No save file found for name: $name")
+            throw IllegalArgumentException("No save file found for name: $name")
         }
 
         return mapper.readValue(save, Settings::class.java)

